internal/resources/google/compute_cost_component_helpers.go

```diff
@@ -64,6 +66,97 @@ func computeCostComponent(region, machineType string, purchaseOption string, ins
 	}
 }
 
+// customComputeCostComponent returns a cost component for custom Compute instance usage.
+func customComputeCostComponents(region, machineType string, purchaseOption string, instanceCount int64, monthlyHours *float64) []*schema.CostComponent {
+	sustainedUseDiscount := 0.0
+	fixPurchaseOption := "Preemptible"
+	if strings.ToLower(purchaseOption) == "on_demand" {
+		fixPurchaseOption = "OnDemand"
+		switch strings.ToLower(strings.Split(machineType, "-")[0]) {
+		case "c2", "n2", "n2d":
+			sustainedUseDiscount = 0.2
+		case "n1", "f1", "g1", "m1":
+			sustainedUseDiscount = 0.3
+		}
+	}
+
+	var re = regexp.MustCompile(`(\D.+)-(\d+)-(\d.+)`)
+	firstTypeInfo := re.ReplaceAllString(machineType, "$1")
+	strCPUAmount := re.ReplaceAllString(machineType, "$2")
+	strRAMAmount := re.ReplaceAllString(machineType, "$3")
+
+	instanceType := ""
+	if firstTypeInfo != "custom" {
+		instanceType = strings.ToUpper(strings.Split(firstTypeInfo, "-")[0])
+	}
+
+	ext := ""
+	if strings.Contains(strRAMAmount, "ext") {
+		ext = "Extended"
+		strRAMAmount = strings.Split(strRAMAmount, "-")[0]
+	}
+
+	numberOfCores, err := strconv.ParseInt(strCPUAmount, 10, 64)
+	if err != nil {
+		return []*schema.CostComponent{}
+	}
+
+	mbOfRAM, err := strconv.ParseInt(strRAMAmount, 10, 64)
+	if err != nil {
+		return []*schema.CostComponent{}
```

Comment:
@aliscott should I return an empty CostComponent array here? Or is there another more appropriate way of delegating this error?

Reply:
I think we can log a warning and return nil here. Similar to what we have here: https://github.com/infracost/infracost/blob/master/internal/providers/terraform/azure/managed_disk.go#L117-L120

Reply:
There we go: https://github.com/infracost/infracost/pull/2004/commits/fadde9b412fcbee9c75d81574f74b2b372c8971f

Reply:
Do I need to document these Terraform resources I'm using somewhere? Please don't hesitate on suggesting the next steps =)

Reply:
Thanks @7onn! I'll review this today.
We keep a list of [supported resources in our documentation site](https://github.com/infracost/docs/blob/master/docs/supported_resources/google.md), so once this is merged I'll update that to remove the note about custom machine types not being supported.